src/main/java/com/ericsson/gerrit/plugins/eiffel/git/CommitInformation.java

```diff
@@ -0,0 +1,116 @@
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gerrit.extensions.restapi.IdString;
+import com.google.gerrit.extensions.restapi.ResourceNotFoundException;
+import com.google.gerrit.extensions.restapi.UnprocessableEntityException;
+import com.google.gerrit.server.project.CommitResource;
+import com.google.gerrit.server.project.CommitsCollection;
+import com.google.gerrit.server.project.ProjectResource;
+import com.google.gerrit.server.project.ProjectsCollection;
+import com.google.inject.Inject;
+
+/**
+ * A class to fetch information of a commit from the repository.
+ *
+ * <strong>Note:</strong> This class uses injects. To use this class the caller must inject it,
+ * otherwise the injection framework cannot instantiate this class properly.
+ *
+ */
+public class CommitInformation {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CommitInformation.class);
+    private final CommitsCollection commitsCollection;
+    private final ProjectsCollection projectsCollection;
+
+    /**
+     * Package private constructor with injection for testing reasons. Injecting directly as field
+     * variables makes it harder to test
+     */
+    @Inject
+    CommitInformation(final CommitsCollection commitsCollection, final ProjectsCollection projectsCollection) {
+        this.commitsCollection = commitsCollection;
+        this.projectsCollection = projectsCollection;
+    }
+
+    /**
+     * Fetches a list of parents SHA from a commit. An empty list will be returned if any error
+     * occurs.
+     *
+     * @param commitId    The SHA of the commit to operate on
+     * @param projectName The name of the project which the commits resides in
+     * @return A list of parents SHAs
+     */
+    public List<String> getParentsSHAs(final String commitId, final String projectName) {
+        final List<RevCommit> parents = getParents(commitId, projectName);
+        final List<String> parentsSHAs = getSHAs(parents);
+        return parentsSHAs;
+    }
+
+    /**
+     * Will return a list of parents for a given commit. If the collections cannot find the project
+     * or commit, an empty list will be returned.
+     */
+    private List<RevCommit> getParents(final String commitId, final String projectName) {
+        List<RevCommit> parents = Collections.emptyList();
+
+        try {
+            parents = getParentsFromCommit(commitId, projectName);
+        } catch (final UnprocessableEntityException e) {
+            final String message = String.format("Cannot find or load the project %s", projectName);
+            LOGGER.error(message, e);
+        } catch (final ResourceNotFoundException e) {
+            final String message = String.format("Cannot find the commit %s", commitId);
```

Comment:
The logger has a built in string formatter, you basicly create a String formatted variable and sends it to a Stirng formatter that also logs. I have not seen this be done like this anywhere else and consistency is important.

This is how it usually looks:
LOGGER.error("Cannot find or load the project {}", projectName, e);
The exception (e) is dangeling, but the LOGGER handles this and logs the exception.
The logger can handle almost anything in the {}, string, int etc, while String.format need specific for %s %d etc depending on type.

A message variable is goot to have if you log and throw a new exception since you may then reuse the message and exception messgaes should be formated before.

Reply:
I know how the logger work but I find it less readable as the last argument may or may not be part of the error message